api/v1/gitrepository_types.go

```diff
@@ -38,6 +38,31 @@ const (
 	IncludeUnavailableCondition string = "IncludeUnavailable"
 )
 
+// GitVerificationMode specifies the verification mode for a Git repository.
+type GitVerificationMode string
+
+// Valid checks the validity of the Git verification mode.
+func (m GitVerificationMode) Valid() bool {
+	switch m {
+	case ModeHEAD, ModeTag, ModeTagAndHEAD:
+		return true
+	default:
+		return false
+	}
+}
+
+const (
+	// ModeHEAD implies that the HEAD of the Git repository (after it has been
+	// checked out to the required commit) should be verified.
+	ModeHEAD GitVerificationMode = "HEAD"
```

Comment:
didn't add one because its not just a plain string but rather a typed string, so the type indicates the fact that these constants are related to Git.  but, if later we want to add something like `ModeTag OCIVerificationMode`, we won't be able to, so yes lets change it to something like `ModeGitTag`?

Reply:
i do indeed think its too long 😬